### PR TITLE
add updateSubgraphSpec utility for nested graph updates

### PR DESCRIPTION
**Pull request by Mbeaulne**
## Description

Added a new `updateSubgraphSpec` utility function that allows updating nested subgraph specifications within a component tree while maintaining immutability. This function recursively navigates through a subgraph path and replaces the target subgraph's spec with an updated version.

## Type of Change

- [x] New feature
- [x] Improvement

## Checklist

- [x] I have tested this does not break current pipelines / runs functionality
- [x] I have tested the changes on staging

## Test Instructions

The implementation includes comprehensive test cases that verify:
- Updating subgraphs at different nesting levels
- Maintaining immutability of the original spec
- Preserving sibling tasks and properties
- Handling edge cases (invalid task IDs, non-subgraph tasks)

Additionally, minor code formatting changes were made to improve readability in `Details.tsx` and `ExecutionDataProvider.tsx`.

### CODE REVIEW

**Comment by Mbeaulne** [comment]
* **#1185** <a href="https://app.graphite.dev/github/pr/Cloud-Pipelines/pipeline-studio-app/1185?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1184** <a href="https://app.graphite.dev/github/pr/Cloud-Pipelines/pipeline-studio-app/1184?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/Cloud-Pipelines/pipeline-studio-app/1184?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `master`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/utils/subgraphUtils.ts [hunk 1]
```diff
@@ -107,3 +107,127 @@ export const getSubgraphComponentSpec = (
...
+  }
+
+  // For nested subgraphs, we need to recursively update the structure
+  return updateSubgraphSpecRecursive(
+    rootSpec,
+    subgraphPath,
+    updatedSubgraphSpec,
+    1,
+  );
+};
```
Comment (camielvs) [inline src/utils/subgraphUtils.ts]: This method feels superficial. It may need a slight adjustment but `updateSubgraphSpecRecursive` should be able to handle everything itself

File: src/utils/subgraphUtils.ts [hunk 2]
```diff
@@ -107,3 +107,127 @@ export const getSubgraphComponentSpec = (
...
+ * @param updatedSubgraphSpec - The updated spec to insert
+ * @param currentDepth - Current depth in the path (1-indexed, skipping "root")
+ * @returns A new ComponentSpec with the update applied
+ */
+const updateSubgraphSpecRecursive = (
+  currentSpec: ComponentSpec,
+  subgraphPath: string[],
+  updatedSubgraphSpec: ComponentSpec,
+  currentDepth: number,
+): ComponentSpec => {
```
Comment (camielvs) [inline src/utils/subgraphUtils.ts]: We don't necessarily need to mash it out here, but off the top of my head it feels like we _could _ remove the `currentDepth` parameter by utilizing `subgraphPath.length`, i.e. by popping out each path as we process our way through it.

File: src/utils/subgraphUtils.ts [hunk 3]
```diff
@@ -107,3 +107,87 @@ export const getSubgraphComponentSpec = (
...
+ *   modifiedSubgraphSpec
+ * );
+ */
+export const updateSubgraphSpec = (
+  currentSpec: ComponentSpec,
+  subgraphPath: string[],
+  updatedSubgraphSpec: ComponentSpec,
+): ComponentSpec => {
+  const path =
+    subgraphPath.length > 0 && subgraphPath[0] === ROOT_TASK_ID
```
Comment (camielvs) [inline src/utils/subgraphUtils.ts:135]:
```suggestion
    subgraphPath[0] === ROOT_TASK_ID
```

I may be wrong but I think this should still work

**Review by maxy-shpfy — APPROVED** [review]
